llvm/lib/Analysis/MemoryBuiltins.cpp

```diff
@@ -50,6 +51,12 @@ using namespace llvm;
 
 #define DEBUG_TYPE "memory-builtins"
 
+static cl::opt<unsigned> ObjectSizeOffsetVisitorMaxRecurseDepth(
+    "object-size-offset-visitor-max-recurse-depth",
+    cl::desc(
+        "Maximum number of PHIs for ObjectSizeOffsetVisitor to look through"),
+    cl::init(100));
```

Comment:
I think this should either not be a depth (but total visited count) or be a lot smaller. Recursing 100 levels allows for a *lot* of fan-out. Especially as this is only counting the phis and not other instructions.

Reply:
Lowered to 20.
I think it's unlikely that we hit huge fanout in real world code, it's typically pretty linear at least with large autogenerated code. In the case I was looking at, this dropped DSE time from >30s to <1s.

Reply:
I've adjusted the call in DSE to check for an identified object first: https://github.com/llvm/llvm-project/commit/7aab12e1c52a5891bed9a594d3f161a8bebf976d